### PR TITLE
Release 0.5.0

### PR DESCRIPTION
**Pull request by wgromniak**
Changes:
* support for inverse instruments,
* settlement hour update,
* allowed negative marings,
* better WebSocket error handling,
* more convenient interfaces.

@muroc, I will be releasing 0.5.0 to Maven Central once you ACK.

### CODE REVIEW

File: src/main/java/net/quedex/api/common/WebsocketStream.java [hunk 1]
```diff
@@ -59,7 +60,12 @@ public void onClose(int code, String reason, boolean remote) {
 
             @Override
             public void onError(Exception ex) {
-                WebsocketStream.this.onError(new CommunicationException("Websocket error", ex));
+                if (ex instanceof IOException && "Broken pipe".equals(ex.getMessage())) {
```
Comment (muroc) [inline src/main/java/net/quedex/api/common/WebsocketStream.java]: Will it work on Windows and Mac OS?

Comment (wgromniak) [reply src/main/java/net/quedex/api/common/WebsocketStream.java]: Good point - quick research suggests that it won't. Maybe we should treat all `IOExceptions` as `DisconnectedException`?

Comment (muroc) [reply src/main/java/net/quedex/api/common/WebsocketStream.java]: I think we should ensure that socket is closed before this method receives `IOException`.

Comment (wgromniak) [reply src/main/java/net/quedex/api/common/WebsocketStream.java]: If be "ensure" you mean:
1. Enforcing this programmatically - this is impossible, we don't have control over the state of the socket and calls of this method.
2. Verifying this in the implementation of the WebSocket library - I've investigated all the places that call this method and it seems that it's used only if a connection is closed or is about to be closed (the next code line following an `onError` closes the connection).

Comment (muroc) [reply src/main/java/net/quedex/api/common/WebsocketStream.java]: 2 is enough (we could implement it ourselves, but this is an implementation detail actually).

**Comment by wgromniak** [comment]
I've overwritten history - the only change is turning all `IOExceptions` into `DisconnectedException` in `onError` in `WebsocketSteam`, commit "Added Disconnect exception...".

**Comment by wgromniak** [comment]
@muroc, thanks.

**Pull request merged**